src/read.rs

```diff
@@ -233,6 +234,46 @@ impl<R: Read + io::Seek> ZipArchive<R> {
         })
     }
 
+    /// Extract a Zip archive into a directory.
+    ///
+    /// Paths are sanitized so that they cannot escape the given directory.
+    ///
+    /// # Platform-specific behaviour
+    ///
+    /// On unix systems permissions from the zip file are preserved, if they exist.
+    pub fn extract(&mut self, directory: &dyn AsRef<Path>) -> ZipResult<()> {
```

Comment:
Is there a reason that directory is a trait object instead of being a generic param?

Reply:
No particular reason. Slight decrease in compile time might be worth the cost of dynamic dispatch but it's really neither here nor there.

Reply:
I personally find it surprising. I would personally like to have the policy of preferring static dispatch unless there's a reason not to. If we can change this, we can merge the PR